salt/reviewer-suggestions/init.sls

```diff
@@ -73,15 +85,39 @@ reviewer-suggestions-configure:
             - reviewer-suggestions-build-essential
             - python-dev
             - reviewer-suggestions-repository
+            - reviewer-suggestions-app-cfg
 
-reviewer-suggestions-preprocess:
+reviewer-suggestions-app-cfg:
+    file.managed:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /srv/reviewer-suggestions/app.cfg
+        - source: 
+            - salt://reviewer-suggestions/config/srv-reviewer-suggestions-{{ salt['elife.cfg']('project.branch') }}.cfg
```

Comment:
you don't have multiple configurations based on branches in this project (complexity), so the second line is enough and the first can be deleted